policy/src/main/java/org/apache/brooklyn/policy/failover/PropagatePrimaryEnricher.java

```diff
@@ -0,0 +1,134 @@
+//    public static final ConfigKey<Map<? extends Sensor<?>, ? extends Sensor<?>>> SENSOR_MAPPING = Propagator.SENSOR_MAPPING;
+
+    // also no longer support effectors
+//    public static final ConfigKey<Boolean> PROPAGATE_EFFECTORS = ConfigKeys.newBooleanConfigKey("propagate.effectors",
+//        "Whether to propagate effectors, default true (effectors already defined here will not be propagated)",
+//        true);
+    
+    public void setEntity(@SuppressWarnings("deprecation") org.apache.brooklyn.api.entity.EntityLocal entity) {
+        super.setEntity(entity);
+        
+        subscriptions().subscribe(entity, Sensors.newSensor(Entity.class, config().get(PRIMARY_SENSOR_NAME)), this);
+        onEvent(null);
+    }
+
+    @Override
+    public synchronized void onEvent(SensorEvent<Object> event) {
+        Entity primary = entity.getAttribute( Sensors.newSensor(Entity.class, config().get(PRIMARY_SENSOR_NAME)) );
+        final Entity lastPrimary = config().get(CURRENT_PROPAGATED_PRODUCER);
+        if (!Objects.equal(primary, lastPrimary)) {
+            log.debug("Removing propagated items from "+lastPrimary+" at "+entity);
+            
+            final Propagator propagator = getManagementContext().lookup(config().get(CURRENT_PROPAGATOR_ID), Propagator.class);
+            // remove propagator
+            if (propagator!=null) {
+                entity.enrichers().remove(propagator);
+                config().set(CURRENT_PROPAGATOR_ID, (String)null);
+            }
+            
+            // remove propagated sensors
+            Collection<? extends Sensor<?>> sensorsToRemove = config().get(PROPAGATING);
+            if (sensorsToRemove!=null) {
+                for (Sensor<?> s: sensorsToRemove) {
+                    if (s instanceof AttributeSensor) {
+                        ((EntityInternal)entity).sensors().remove((AttributeSensor<?>)s);
+                    }
+                }
+            }            
+            
+            if (primary!=null) {
+                config().set(CURRENT_PROPAGATED_PRODUCER, primary);
+                
+                // add propagator
+                addPropagatorEnricher(primary);
+            }
+        }
+    }
+    
+    @Override
+    protected <T> void doReconfigureConfig(ConfigKey<T> key, T val) {
+        if (CURRENT_PROPAGATED_PRODUCER.equals(key)) return;
+        if (CURRENT_PROPAGATOR_ID.equals(key)) return;
+        // disallow anything else
+        super.doReconfigureConfig(key, val);
+    }
+
+    protected void addPropagatorEnricher(Entity primary) {
+        EnricherSpec<Propagator> spec = EnricherSpec.create(Propagator.class);
+        spec.configure(Propagator.PRODUCER, primary);
+        
+        Collection<? extends Sensor<?>> sensorsToPropagate = getConfig(PROPAGATING);
+        if (sensorsToPropagate==null || sensorsToPropagate.isEmpty()) {
+            log.warn("");
```

Comment:
Missing log.warn message. 

TL;DR ignore this next comment :-) If we want to do this check and warn, I'd be tempted to do it in `setEntity()` so the user gets an immediate warning, rather than waiting until the primary is set. However, I'm also hesitant to resolve config there because if one uses DSL / deferred suppliers then it could fail if the value is not yet available.